### ISSUE TITLE
Tests: distinguish by the need of network connection

### ISSUE THREAD

**Issue by nico202** [Nheko-Reborn/mtxclient#22]
Hi, I'm building this library on guix.  The build environment does not have network connection.  I'd still like to run as many tests as possible, so it would be useful to have a flag that distinguish between those that depends on a working network connection and those that can be run locally.

Thanks!

**Comment by nico202** [comment]
Those are failing for me
- BasicConnectivity :: freeze
- ClientAPI :: freeze
- MediaAPI :: media_api.cpp:49: err: actual true, expected false
- Encryption :: freeze

**Comment by deepbluev7** [comment]
Yes, this is also a big issue for me, since I don't have docker on my laptop, so I can't run the network tests there. We should definitely do that!

**Comment by deepbluev7** [comment]
If you use meson to build mtxclient, there are now 2 suites. nonetwork should be able to run without network. There might be some tests that are excluded, that could be run without network, but are not in that suite and we don't test it without network that often though, but it should be a first step.

https://github.com/Nheko-Reborn/mtxclient/blob/master/tests/meson.build
